Platform security 20230919

pyth0n1c opened the pull request:
Detections for Platform Security

pyth0n1c committed:
Initial commit of draft searches
 before adding datasets

pyth0n1c committed:
associated datasets with detections

pyth0n1c committed:
Fix missing macro

pyth0n1c committed:
fix search typos

pyth0n1c committed:
fix dataset links

pyth0n1c committed:
fix search formatting

pyth0n1c committed:
Merge branch 'develop' into platform_security_20230919

pyth0n1c committed:
Search improvements

pyth0n1c committed:
Add timing to searches

pyth0n1c committed:
fix search and notable

pyth0n1c committed:
Parse out fields for runshellscript search

srv-rr-gh-researchbt committed:
Branch was auto-updated.

srv-rr-gh-researchbt committed:
Branch was auto-updated.

srv-rr-gh-researchbt committed:
Branch was auto-updated.

patel-bhavin commented on detections/application/splunk_absolute_path_traversal_using_runshellscript.yml:
```diff
@@ -0,0 +1,59 @@
+status: production
+type: Hunting
+data_source:
+- python.log
```
this should be empty

patel-bhavin commented on detections/application/splunk_absolute_path_traversal_using_runshellscript.yml:
Lets rename `splunk_server` to `dest` and update this on observables

patel-bhavin replied:
ignore

patel-bhavin commented on detections/application/splunk_cmd_injection_using_external_lookups.yml:
```diff
@@ -0,0 +1,49 @@
+data_source:
+- audittrial
+description: In Splunk Enterprise versions below 8.2.12, 9.0.6, and 9.1.1, an attacker can create an external lookup that calls a legacy internal function. The attacker can use this internal function to insert code into the Splunk platform installation directory. From there, a user can execute arbitrary code on the Splunk platform Instance.
+search: '`audit_searches` search = "| *runshellscript* """ | stats count min(_time) as firstTime max(_time) as lastTime by search user splunk_server 
```
Lets rename `host` to `dest` and update this on observables and message and this search is currently failing CI

patel-bhavin replied:
ignore

pyth0n1c replied:
After followup conversation, we will keep this as splunk_server for these detections.

patel-bhavin commented on detections/application/splunk_dos_using_malformed_saml_request.yml:
```diff
@@ -0,0 +1,47 @@
+data_source:
+- splunkd.log 
+description: In Splunk Enterprise versions lower than 9.0.6, and 8.2.12, an attacker can send a malformed security assertion markup language SAML request to the /saml/acs REST endpoint which can cause a denial of service through a crash or hang of the Splunk daemon.The SAML extensible markup language (XML) parser does not fail SAML signature validation when the attacker modifies the URI in the SA ...
+search: '`splunkd` event_message=*error* expr=*xpointer* | stats count min(_time) as firstTime max(_time) as lastTime by component expr host event_message | `security_content_ctime(firstTime)`| `security_content_ctime(lastTime)` | `splunk_dos_using_malformed_saml_request_filter`'
```
Lets rename host to dest and update this on observables and message

patel-bhavin commented on detections/application/splunk_reflected_xss_on_app_search_table_endpoint.yml:
i dont see a host related field in the SPL, can we perhaps add that? and then rename host to dest

srv-rr-gh-researchbt committed:
Branch was auto-updated.

srv-rr-gh-researchbt committed:
Branch was auto-updated.

srv-rr-gh-researchbt committed:
Branch was auto-updated.

srv-rr-gh-researchbt committed:
Branch was auto-updated.

pyth0n1c committed:
Minor updates to improve notables
and result rendering.

pyth0n1c committed:
Delete detection and migrate
logic and documentation to risky
command lookup file. Update macro
to point to new risky command csv.
Fiox some opservables.

pyth0n1c committed:
add another filter so that we do not flag
scripts running out of c:

srv-rr-gh-researchbt committed:
Branch was auto-updated.

srv-rr-gh-researchbt committed:
Branch was auto-updated.

srv-rr-gh-researchbt committed:
Branch was auto-updated.

pyth0n1c committed:
Added fillnull to improve detection accuracy for all logs.

srv-rr-gh-researchbt committed:
Branch was auto-updated.

pyth0n1c committed:
First version of new detection for testing

pyth0n1c committed:
Merge branch 'platform_security_20230919' of https://github.com/splunk/security_content into platform_security_20230919

pyth0n1c committed:
Update to new search after dynamic testing and refinement.

pyth0n1c committed:
remove source= from search

pyth0n1c committed:
use macro instead of using index and sourcetype directly.

pyth0n1c committed:
properly format times in risky command search
